Not retrying after some errors

After the following error occurred,

```
[2015-01-21 02:47:22.212] NIO [ERROR] [Instagram.Insta] GET request failed, details: ('Connection aborted.', BadStatusLine("''",))
```

no retry happened and the block became dead without reporting an error state. We should be retrying after errors such as this one.

--- a/rest/rest_block.py
+++ b/rest/rest_block.py
@@ -115,12 +115,8 @@ class RESTPolling(Block):
         # Increment the number of lock waiters so we don't build up too many
         self._num_locks += 1
         with self._poll_lock:
-            if self._retry_job is None:
-                self._locked_poll(paging)
-            else:
-                self._logger.debug(
-                    "Skipping this poll since a retry job is scheduled"
-                )
+            self._locked_poll(paging)
+
         self._num_locks -= 1
 
     def _locked_poll(self, paging=False):
@@ -135,7 +131,7 @@ class RESTPolling(Block):
             "{}: {}".format("Paging" if paging else "Polling", url)
         )
 
-        resp = self._execute_request(url, headers)
+        resp = self._execute_request(url, headers, paging)
         if resp is None:
             return
 
@@ -450,7 +446,7 @@ class RESTPolling(Block):
         m = re.match(exp, date)
         return datetime(*[int(n) for n in m.groups(0)])
 
-    def _execute_request(self, url, headers):
+    def _execute_request(self, url, headers, paging):
         """ Execute the request, accounting for possible errors
 
         """

--- a/rest/tests/test_rest_block.py
+++ b/rest/tests/test_rest_block.py
@@ -135,6 +135,33 @@ class TestRESTPolling(NIOBlockTestCase):
 
         blk.stop()
 
+    @patch("requests.get", side_effect=Exception("mock get fail"))
+    @patch.object(RESTPolling, "_authenticate")
+    @patch.object(RESTPolling, "_retry_poll")
+    def test_get_error_retry(self, mock_retry, mock_auth, mock_get):
+        es = [Event(), Event()]
+        blk = RESTRetry(es)
+        self.configure_block(blk, {
+            "log_level": "DEBUG",
+            "polling_interval": {
+                "seconds": 1
+            },
+            "retry_interval": {
+                "seconds": 1
+            },
+            "queries": [
+                "foobar"
+            ]
+        })
+        blk.start()
+        es[1].wait(2)
+
+        self.assertEqual(mock_auth.call_count, 2)
+        self.assertEqual(mock_retry.call_count, 1)
+        self.assertEqual(mock_get.call_count, 1)
+
+        blk.stop()
+
     @patch("requests.get")
     @patch.object(RESTPolling, "_process_response")
     @patch.object(RESTPolling, "_prepare_url")

